### ISSUE TITLE
[BUG] ANT+ Speed Conversion

### ISSUE THREAD

**Issue by ptrinephi** [cagnulein/qdomyos-zwift#167]
Beta 2.3.11.  It appears the speed transmitted via ANT+ from QZ is incorrect.  
When testing with ipSensorMan the app shows 64.2m/s (I'm guessing this is meter per second) which would convert to 143mph!  
From BKOOL, it reports a speed 4 times less than QZ.

![2021-02-07_19-21-03](https://user-images.githubusercontent.com/6604478/107172251-b430fa00-6979-11eb-919a-b33d8a5fe8d7.png)
![2021-02-07_19-21-53](https://user-images.githubusercontent.com/6604478/107172287-d296f580-6979-11eb-9de6-1a76e6af89de.png)

**Comment by cagnulein** [comment]
On ant+ speed is sending without unit, you have to calibrate it in the app/
watch that you are using, configuring the wheel ratio.

Il lun 8 feb 2021, 04:24 3phi <notifications@github.com> ha scritto:

**Comment by ptrinephi** [comment]
Are you sure?  IpSensorMan and BKool both report the same value.

![20210208_103232.jpg](https://user-images.githubusercontent.com/6604478/107265350-05340300-69f9-11eb-9c9c-66fd9aa00a81.jpg)

**Comment by cagnulein** [comment]
Yes same value because the default ratio i guess is 0.1
Speed for ant+ is just cadence multiplied for the wheel size

**Comment by ptrinephi** [comment]
Ok, I emailed BKool support.

**Comment by cagnulein** [comment]
Bkool doesn't have settings?

Il lun 8 feb 2021, 19:40 3phi <notifications@github.com> ha scritto:

**Comment by ptrinephi** [comment]
Not that I can tell but reading their help doc it says you need to calibrate your BKOOL bike.  Maybe Qz need an option to convert the values sent?

**Comment by cagnulein** [comment]
I should read the ant documentation. I didn't yet. Anyway on my Watch there
is a setting to calibrate the wheel. This is why I say that the value on
ant+ is unitless

Il lun 8 feb 2021, 20:57 3phi <notifications@github.com> ha scritto:

**Comment by ptrinephi** [comment]
For someone who hasn't read the doc you're doing a pretty damn good job!

From my quick reading, it looks like the app of the ANT+ device is where you calculate the speed which is what your watch is I suppose.  So in Qz's case, Qz is the app that needs the calculation.  Since Qz already knows the speed, I assume it should be a fairly easy math to get the ANT+ speed value which looks like it's meter per second to take your km/h divide by 3.6 will get your meter per second and the display can do the mph conversion as usual.

**Comment by cagnulein** [comment]
Ok but I don't understand so what value should I put on my Watch as wheel
radius. Tomorrow i will think about it ;)

Il lun 8 feb 2021, 21:20 3phi <notifications@github.com> ha scritto:

**Comment by ptrinephi** [comment]
Isn't your watch connected to an external speed sensor?  Once connected you have to set the wheel size of your bike which is used to calculate the speed.  Since we use spin bikes there's no real wheel size so I think you can just use the speed that Qz already knows and calculate the m/s from there.  Can you link your watch to ipSensorMan app and see what speed it reports on it?

**Comment by cagnulein** [comment]
Yes i Will do today ;)

Il lun 8 feb 2021, 22:10 3phi <notifications@github.com> ha scritto:

**Comment by cagnulein** [comment]
now i understood: on my watch i have an automatic setting and a manual setting for the wheel. I guess ipSensorMan uses the "automatic" one. Anyway i didn't fix the stability of the speed yet (i hoped yesterday, but probably it's not just math). So i will have to debug it more, but not today ;)

**Comment by ptrinephi** [comment]
Beta 2.3.12, looks like you did something here but now the speed is still wrong.  When Qz displays 20mph ipSensorMan displays 0.1-0.4

**Comment by cagnulein** [comment]
Yeah i know, I didn't fix it yet

Il mar 9 feb 2021, 18:19 3phi <notifications@github.com> ha scritto:

**Comment by mschwaberow** [comment]
Same issue here.  I messed around with the wheel size for the Speed sensor in my Garmin Fenix 6 Pro settings. I tried Automatic, then Manual with 2000mm and 500mm wheel sizes just to get a sense of how to calibrate it.  All three reported incorrect low speeds with no real difference.  Hence, I don't think the calibration setting on the watch is the issue.

Thanks again for this feature - very excited to have it fully operational!!
